wallet-new/src/Cardano/Wallet/WalletLayer/Kernel/Transactions.hs

```diff
@@ -0,0 +1,162 @@
+module Cardano.Wallet.WalletLayer.Kernel.Transactions (
+    getTransactions
+) where
+
+import           Universum
+
+import qualified Cardano.Wallet.Kernel as Kernel
+import qualified Cardano.Wallet.Kernel.DB.HdWallet as HD
+import           Cardano.Wallet.Kernel.DB.InDb (InDb (..))
+import           Cardano.Wallet.Kernel.DB.TxMeta (TxMeta (..))
+import qualified Cardano.Wallet.Kernel.DB.TxMeta as TxMeta
+import           Cardano.Wallet.WalletLayer.Types (GetTxError (..))
+
+import           Cardano.Wallet.API.Indices
+import           Cardano.Wallet.API.Request
+import qualified Cardano.Wallet.API.Request.Filter as F
+import           Cardano.Wallet.API.Request.Pagination
+import qualified Cardano.Wallet.API.Request.Sort as S
+import           Cardano.Wallet.API.Response
+import           Cardano.Wallet.API.V1.Types (V1 (..), unV1)
+import qualified Cardano.Wallet.API.V1.Types as V1
+import           GHC.TypeLits (symbolVal)
+import           Pos.Core as Core
+
+getTransactions :: MonadIO m
+                => Kernel.PassiveWallet
+                -> Maybe V1.WalletId
+                -> Maybe V1.AccountIndex
+                -> Maybe (V1 Address)
+                -> RequestParams
+                -> FilterOperations V1.Transaction
+                -> SortOperations V1.Transaction
+                -> m (Either GetTxError (WalletResponse [V1.Transaction]))
+getTransactions wallet mbWalletId mbAccountIndex mbAddress params fop sop = liftIO $ do
```

Comment:
If you take a look at the other functions in `WalletLayer.Kernel.*`, you can see how I refactored those. Can I suggest you do the same here, to keep the code clean and consistent?

Reply:
(I introduced that after you worked on this, so entirely my own fault :slightly_smiling_face:)

Reply:
I think this doesn`t apply because getTransactions can`t be pure even with a snapshot: it still talks with the Meta db.

Reply:
It doesn't have to me, most of those functions aren't pure.

Reply:
You'd have a `runExceptT` at the very top, and then use `ExceptT` to do the error threading and the underlying monad for whatever effects it needs.

Reply:
Also I think the order of reading is important from the 2 dbs (adding a TODO comment for this).

Reply:
Yes, I'm not suggesting to make it pure. I'm merely suggesting to use `ExceptT` to take care of the error threading.

Reply:
Right, working on it. I thought you meant the `read only` refactoring in `WalletLayer/Kernel.hs`.